### PR TITLE
feat: Bump React 19 to the stable release version (#106)

### PR DESCRIPTION
**Pull request by soniaklimas**
I want to merge this change because it copies PDP improvements from fashion branch and adjust them to Nimara code.

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: apps/storefront/src/app/[locale]/(main)/products/[slug]/components/details-dropdown.tsx
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/products/[slug]/components/details-dropdown.tsx:1]: An empty file?

File: apps/storefront/src/app/[locale]/(main)/products/[slug]/components/attributes-dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            <AccordionContent>
+              {attribute.values.map((val) => {
+                if (val.richText) {
+                  return (
+                    <RichText key={val.name} jsonStringData={val.richText} />
+                  );
+                }
+
+                return <p key={val.name}>{val.name}</p>;
+              })}
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/products/[slug]/components/attributes-dropdown.tsx:45]:
```suggestion
              {attribute.values.map((val) => val.richText ? (
                    <RichText key={val.name} jsonStringData={val.richText} />
                  ) : ( <p key={val.name}>{val.name}</p> )
              )}
```

File: apps/storefront/src/app/[locale]/(main)/products/[slug]/components/product-display.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  // const imagesToDisplay = chosenVariant?.images?.length
+  //   ? chosenVariant.images
+  //   : looselyMatchingVariants.length > 0
+  //     ? looselyMatchingVariants
+  //         .flatMap(({ images }) => images)
+  //         .filter(
+  //           (image, index, self) =>
+  //             index === self.findIndex((i) => i.url === image.url),
+  //         )
+  //     : product.images;
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/products/[slug]/components/product-display.tsx:71]: unused code?

File: apps/storefront/src/app/[locale]/(main)/products/[slug]/components/product-display.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+  const imagesToDisplay = getImagesToDisplay({
+    chosenVariant,
+    looselyMatchingVariants,
+    productImages: images,
+  });
+
+  console.log("imagesToDisplay", imagesToDisplay);
+  console.log("images", images);
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/products/[slug]/components/product-display.tsx:80]: to remove?

File: packages/domain/src/objects/Product.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import type { Attribute } from "./Attribute";
 import type { Image, TaxedPrice } from "./common";
+import { type SearchProduct } from "./SearchProduct";
```
Comment (karolkarolka) [inline packages/domain/src/objects/Product.ts:3]:
```suggestion
import type { SearchProduct } from "./SearchProduct";
```

File: packages/ui/src/components/alert.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    ref={ref}
+    role="alert"
+    className={cn(alertVariants({ variant }), className)}
+    {...props}
+  />
+));
+
+Alert.displayName = "Alert";
+
+const AlertTitle = React.forwardRef<
```
Comment (karolkarolka) [inline packages/ui/src/components/alert.tsx:36]: remove all `forwardRef`, it isn't supported in React 19, check for types for `ref` prop in other components, it should be smth like `MutableObject`

File: apps/storefront/src/app/[locale]/(main)/products/[slug]/components/variant-selector-utils.ts [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+          }
+        }
+      },
+    );
+  });
+
+  return allSelectionAttributes;
+};
+
+// ! currently unsued
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/products/[slug]/components/variant-selector-utils.ts:161]: it can be shipped to the `main` branch with such comment, delete the function if unused

**Review by karolkarolka — CHANGES_REQUESTED** [review]
I can see the changes for search page so adjust the PR name, please.

File: apps/storefront/src/app/[locale]/(main)/products/[slug]/components/product-display.tsx [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+              product={product}
+              productAvailability={availability}
+              user={user}
+            />
+
+            {hasFreeShipping && (
+              <Alert>
+                <Truck className="size-4" />
+                <AlertTitle>Free shipping</AlertTitle>
+
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/products/[slug]/components/product-display.tsx:144]: unnecessary new line

File: apps/storefront/src/app/[locale]/(main)/products/[slug]/components/product-display.tsx [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+            )}
+
+            <AttributesDropdown attributes={attributesToDisplay} />
+          </section>
+        </div>
+      </div>
+
+      {product.relatedProducts.length > 0 && (
+        <div className="mb-7 mt-10 md:mb-14 md:mt-20">
+          <h2 className="mb-4 text-4xl text-black">You may also like</h2>
```
Comment (karolkarolka) [inline apps/storefront/src/app/[locale]/(main)/products/[slug]/components/product-display.tsx:163]: Ah now I can see that all copies on PDP are lacking the i18n wrapper, please wrap them with translate function

**Review by karolkarolka — CHANGES_REQUESTED** [review]

**Review by karolkarolka — APPROVED** [review]
